### PR TITLE
[Fix-16650] Memory leaks in KubernetesApplicationManager

### LINKED ISSUE #16650: [Bug] [plugin] Memory leakage occurs in k8s task
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

`KubernetesApplicationManager` is used to hold the k8s client when k8s task is running, but the cache is only removed in `KubernetesApplicationManager#killApplication` method, which is invoked when k8s task is killed or throw an exception, or failover happens. But when k8s task is finished successfully, the method is not invoked, caused the cached is not removed.

### What you expected to happen

The cached should be removed when k8s task finished successfully

### How to reproduce

create and execute a k8s task, and periodically print the content of `KubernetesApplicationManager` in the meantime. When k8s task succeeded, the cache of this task still exists.

### Anything else

_No response_

### Version

3.2.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ChaoquanTao**
## Purpose of the pull request

This PR resolves the memory leakage in KubernetesApplicationManager

- close #16650

## Brief change log

- *Add `removeK8sClientCache` method in ProcessUtils*
- *remove K8s client cache when k8s job is done*

## Verify this pull request

This pull request is code cleanup without any test coverage.

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

### CODE REVIEW

**Comment by Gallardot** [comment]
@Radeity PTAL.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java [hunk 1]
```diff
@@ -243,4 +243,10 @@ public static LogWatch getPodLogWatcher(K8sTaskExecutionContext k8sTaskExecution
                 .getPodLogWatcher(
                         new KubernetesApplicationManagerContext(k8sTaskExecutionContext, taskAppId, containerName));
     }
+
+    public static void removeK8sClientCache(String taskAppId) {
+        KubernetesApplicationManager applicationManager =
+                (KubernetesApplicationManager) applicationManagerMap.get(ResourceManagerType.KUBERNETES);
+        applicationManager.removeCache(taskAppId);
+    }
```
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java:251]: Please add some UT for this.

Comment (ChaoquanTao) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java:251]: Done

**Review by SbloodyS — APPROVED** [review]
LGTM

**Review by Gallardot — APPROVED** [review]

**Pull request merged**